### PR TITLE
Split the main fastly interface of fastly.wit

### PR DESCRIPTION
**Pull request by elliottt**
Split `fastly.wit` into different interfaces that follow the existing module separation of the core wasm host imports.

### CODE REVIEW

File: runtime/js-compute-runtime/fastly.wit [hunk 1]
```diff
@@ -553,17 +334,293 @@ interface fastly {
...
+/*
+ * Fastly HTTP Req
+ */
+interface http-req {
+
+  use types.{
+    request-handle, http-cache-override-tag, http-version, body-handle, error,
+    response, pending-request-handle, content-encodings, framing-headers-mode,
+    dynamic-backend-config
+  }
```
Comment (guybedford) [inline runtime/js-compute-runtime/fastly.wit:422]: When it comes to type sharing, the WASI approach is to have a subsystem specific interface - eg:

```
interface http {
  type request-handle = u32 // etc
}

interface http-req {
  use http.{request-handle}
}
```

Or even call it `http-types` directly.

**Review by guybedford — COMMENTED** [review]
Very nice work, just a couple of notes.

File: runtime/js-compute-runtime/fastly.wit [hunk 2]
```diff
@@ -553,17 +334,293 @@ interface fastly {
...
     must-insert-or-update,
   }
+}
+
+/*
+ * Fastly Log
+ */
+interface log {
+
+  use types.{error, log-endpoint-handle}
```
[the remark this answers is not in the document]
Comment (elliottt) [reply runtime/js-compute-runtime/fastly.wit:344]: Happy to move types only used in a single interface into those interfaces. That's how I had it originally, but decided somewhat arbitrarily to move everything into the types interface.

File: runtime/js-compute-runtime/fastly.wit (hunk 1 above)
Comment (elliottt) [reply runtime/js-compute-runtime/fastly.wit:422]: That works for me!

**Review by guybedford — APPROVED** [review]
👍

**Pull request merged**